gr-blocks/lib/selector_impl.cc

```diff
@@ -122,11 +123,16 @@ void selector_impl::handle_msg_output_index(const pmt::pmt_t& msg)
 void selector_impl::handle_enable(const pmt::pmt_t& msg)
 {
     if (pmt::is_bool(msg)) {
-        bool en = pmt::to_bool(msg);
+        const bool en = pmt::to_bool(msg);
+        gr::thread::scoped_lock l(d_mutex);
+        d_enabled = en;
+    } else if (pmt::is_pair(msg)) {
+        const bool en = pmt::to_bool(pmt::cdr(msg));
         gr::thread::scoped_lock l(d_mutex);
         d_enabled = en;
     } else {
-        d_logger->warn("handle_enable: Non-PMT type received, expecting Boolean PMT");
+        d_logger->warn(
+            "handle_enable: Non-PMT type received, expecting Boolean PMT or PDU");
```

Comment:
The message is not yours, but it's still wrong (I mean, you can *only* send PMTs around, what they meant to say is that it's an *invalid* PMT message). If you feel like it, you might just fix that :)